webdriver/tests/bidi/browsing_context/context_destroyed/context_destroyed.py

```diff
@@ -81,34 +81,38 @@ async def on_event(_, data):
 async def test_navigate_iframe(
     bidi_session, wait_for_event, wait_for_future_safe, subscribe_events, new_tab, inline, domain
 ):
-    await subscribe_events([CONTEXT_DESTROYED_EVENT])
-
-    on_entry = wait_for_event(CONTEXT_DESTROYED_EVENT)
-
     frame_url = inline("<div>foo</div>")
     url = inline(f"<iframe src='{frame_url}'></iframe>")
     await bidi_session.browsing_context.navigate(
         url=url, context=new_tab["context"], wait="complete"
     )
 
-    contexts = await bidi_session.browsing_context.get_tree(root=new_tab["context"])
-    frame = contexts[0]["children"][0]
+    # Assert that the iframe was created.
+    contexts = await bidi_session.browsing_context.get_tree(
+        root=new_tab["context"])
+    assert len(contexts) == 1
+    assert len(contexts[0]["children"]) == 1
+
+    # Track `browsingContext.contextDestroyed` events.
+    context_destroyed_events = []
+
+    async def on_event(_, data):
+        context_destroyed_events.append(data)
+
+    remove_listener = bidi_session.add_event_listener(CONTEXT_DESTROYED_EVENT,
+                                                      on_event)
+    await subscribe_events([CONTEXT_DESTROYED_EVENT])
 
-    # Navigate to destroy iframes
-    url = inline(f"<iframe src='{frame_url}'></iframe>", domain=domain)
+    # Navigate to another page to destroy iframe.
+    url = inline(f"<div>another page</div>", domain=domain)
     await bidi_session.browsing_context.navigate(
         url=url, context=new_tab["context"], wait="complete"
     )
 
-    context_info = await wait_for_future_safe(on_entry)
+    # Assert the destroyed context was not emitted for the iframe.
```

Comment:
I don't understand these changes. We are subscribing the event on the top-level browsing context and as such should get an event because we navigate away. Why should it be empty without any specific filtering?

Also the test name doesn't really reflect what we are testing here. Maybe we should have two tests:

1. Navigate away top-level 
2. Navigate away the frame

In both cases we should get a single event.